src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -1565,3 +1583,13 @@ int sd_dhcp_server_set_relay_agent_information(
         free_and_replace(server->agent_remote_id, remote_id_dup);
         return 0;
 }
+
+int sd_dhcp_server_set_lease_file(sd_dhcp_server *server, const char *path) {
+        assert_return(server, -EINVAL);
+        assert_return(!sd_dhcp_server_is_running(server), -EBUSY);
+
+        if (!path || !path_is_safe(path))
```

Comment:
This condition is wrong: it should be `if (path && !path_is_safe(path))`, because we want to allow NULL to unset the lease file.